lib/parser.dart

```diff
@@ -192,7 +196,9 @@ class MappingBundle extends Mapping {
   void addMapping(SingleMapping mapping) {
     // TODO(jacobr): verify that targetUrl is valid uri instead of a windows
     // path.
-    _mappings[mapping.targetUrl] = mapping;
+    var targetUrl = ArgumentError.checkNotNull<String>(
```

Comment:
Why do we need the `<String>` here?

Reply:
It isn't inferring the type correctly otherwise in the CFE, I get a `String?` still.

I can file something and add a TODO here.

Reply:
https://github.com/dart-lang/sdk/issues/42227